src/runtime.js

```diff
@@ -196,6 +196,11 @@ var Runtime = {
 #if ASSERTIONS
       assert(args.length == sig.length-1);
 #endif
+      if (Module['usingWasm']) {
+        // look up function by name and call it directly
+        name = Module['indirectFunctions'][ptr]
+        return Module[name].apply(null, args);
```

Comment:
So you're saying it could be something like
`Module['asm'].dynCall_vii(arg1, arg2)` should be generated by emscripten? Even that still has to look up 'dynCall_vii' on the export object just as if it were `Module['asm']['dynCall_vii']`? or are you saying that `Module['dynCall_vii'].apply(...)` is faster than `name = ...; Module[name].apply(...)` ?